src/cast_core/src/actors/CrudActor.cpp

```diff
@@ -0,0 +1,615 @@
+            rhs.read_preference(readPreference);
+        }
+        if (node["BypassDocumentValidation"]) {
+            auto bypassValidation = node["BypassDocumentValidation"].as<bool>();
+            rhs.bypass_document_validation(bypassValidation);
+        }
+        if (node["Hint"]) {
+            auto h = node["Hint"].as<std::string>();
+            auto hint = mongocxx::hint(h);
+            rhs.hint(hint);
+        }
+        if (node["WriteConcern"]) {
+            auto wc = node["WriteConcern"].as<mongocxx::write_concern>();
+            rhs.write_concern(wc);
+        }
+        return true;
+    }
+};
+
+template <>
+struct convert<mongocxx::options::bulk_write> {
+    using BulkWriteOptions = mongocxx::options::bulk_write;
+    static Node encode(const BulkWriteOptions& rhs) {
+        Node node;
+        auto bypassDocValidation = rhs.bypass_document_validation();
+        if (bypassDocValidation) {
+            node["BypassDocumentValidation"] = *bypassDocValidation;
+        }
+        auto isOrdered = rhs.ordered();
+        node["Ordered"] = isOrdered;
+        return node;
+    }
+
+    static bool decode(const Node& node, BulkWriteOptions& rhs) {
+        if (!node.IsMap()) {
+            return false;
+        }
+        if (node["BypassDocumentValidation"]) {
+            auto bypassDocValidation = node["BypassDocumentValidation"].as<bool>();
+            rhs.bypass_document_validation(bypassDocValidation);
+        }
+        if (node["Ordered"]) {
+            auto isOrdered = node["Ordered"].as<bool>();
+            rhs.ordered(isOrdered);
+        }
+        if (node["WriteConcern"]) {
+            auto wc = node["WriteConcern"].as<mongocxx::write_concern>();
+            rhs.write_concern(wc);
+        }
+        return true;
+    }
+};
+
+template <>
+struct convert<mongocxx::options::count> {
+    using CountOptions = mongocxx::options::count;
+    static Node encode(const CountOptions& rhs) {
+        Node node;
+        auto h = rhs.hint();
+        if (h) {
+            auto hint = h->to_value().get_utf8().value;
+            node["Hint"] = std::string(hint);
+        }
+        auto limit = rhs.limit();
+        if (limit) {
+            node["Limit"] = *limit;
+        }
+        auto maxTime = rhs.max_time();
+        if (maxTime) {
+            auto maxTimeSpec = genny::TimeSpec(*maxTime);
+            node["MaxTimeMillis"] = maxTimeSpec.count();
+        }
+        return node;
+    }
+
+    static bool decode(const Node& node, CountOptions& rhs) {
+        if (!node.IsMap()) {
+            return false;
+        }
+        if (node["Hint"]) {
+            auto h = node["Hint"].as<std::string>();
+            auto hint = mongocxx::hint(h);
+            rhs.hint(hint);
+        }
+        if (node["Limit"]) {
+            auto limit = node["Limit"].as<int>();
+            rhs.limit(limit);
+        }
+        if (node["MaxTimeMillis"]) {
+            auto maxTime = node["MaxTimeMillis"].as<int>();
```

Comment:
Per genny's conventions of having time-based units be a first-class concept, you'll want to have this be a `TimeSpec` rather than an `int`. See my earlier comment for handling `wtimeout`.